### ISSUE TITLE
LoadError: ccall method definition: argument 1 type dosen't correspond to a C type

### ISSUE THREAD

**Issue by Blumenkranz** [JuliaParallel/PETSc.jl#187]
I tried to install PETSc with `]add PETSc` and I got
```
ERROR: LoadError: ccall method definition: argument 1 type doesn't correspond to a C type
Stacktrace:
 [1] top-level scope
   @ ~/.julia/packages/PETSc/LSdRz/src/ksp.jl:199
 [2] eval(m::Module, e::Any)
   @ Core ./boot.jl:370
 [3] top-level scope
   @ ~/.julia/packages/PETSc/LSdRz/src/lib.jl:120
 [4] include(mod::Module, _path::String)
   @ Base ./Base.jl:457
 [5] include(x::String)
   @ PETSc ~/.julia/packages/PETSc/LSdRz/src/PETSc.jl:1
 [6] top-level scope
   @ ~/.julia/packages/PETSc/LSdRz/src/PETSc.jl:34
 [7] include
   @ ./Base.jl:457 [inlined]
 [8] include_package_for_output(pkg::Base.PkgId, input::String, depot_path::Vector{String}, dl_load_path::Vector{String}, load_path::Vector{String}, concrete_deps::Vector{Pair{Base.PkgId, UInt128}}, source::Nothing)
   @ Base ./loading.jl:2049
 [9] top-level scope
   @ stdin:3
```
The Julia version is 1.9 and I'm using an M1 mac.

**Comment by boriskaus** [comment]
yes, `PETSc.jl` is not working and not maintained at the moment. I'm trying to halfway maintain the binarybuilder version of PETSc (`PETSc_jll`) but lack time to do this for `PETSc.jl`

**Comment by Blumenkranz** [comment]
I see. Thanks for your reply

**Comment by mSamiolo** [comment]
Noted, just out completition, it does not work also with Julia 1.10.1 with x86 instruction set. Neither if I use an existing PETSc installation.

**Comment by ViralBShah** [comment]
This should be fixed now on the `main` branch. Can you dev the package and try it out?

**Comment by ViralBShah** [comment]
Keeping it open until a new release is tagged.

**Comment by Blumenkranz** [comment]
> This should be fixed now on the `main` branch with #194. Can you dev the package and try it out?

Yes, the main branch can be successfully loaded now
